signing/multisig/blsMultisig.go

```diff
@@ -1,332 +1,82 @@
-	}
-
-	sizeConsensus := uint16(len(pubKeys))
-	sigShares := make([][]byte, sizeConsensus)
-	pk, err := convertStringsToPubKeys(pubKeys, bms.keyGen)
-
-	if err != nil {
-		return err
-	}
-
-	bms.mutSigData.Lock()
-	defer bms.mutSigData.Unlock()
-
-	privKey := bms.data.privKey
-
-	data := &blsMultiSigData{
-		pubKeys:   pk,
-		privKey:   privKey,
-		ownIndex:  index,
-		sigShares: sigShares,
-	}
-
-	bms.data = data
-
-	return nil
-}
-
-// Create generates a multiSigner and initializes corresponding fields with the given params
-func (bms *blsMultiSigner) Create(pubKeys []string, index uint16) (crypto.MultiSigner, error) {
-	bms.mutSigData.RLock()
-	privKey := bms.data.privKey
-	bms.mutSigData.RUnlock()
-
-	return NewBLSMultisig(bms.llSigner, pubKeys, privKey, bms.keyGen, index)
-}
-
 // CreateSignatureShare returns a BLS single signature over the message
-func (bms *blsMultiSigner) CreateSignatureShare(message []byte, _ []byte) ([]byte, error) {
-	bms.mutSigData.Lock()
-	defer bms.mutSigData.Unlock()
-
-	data := bms.data
-	sigShareBytes, err := bms.llSigner.SignShare(data.privKey, message)
+func (bms *blsMultiSigner) CreateSignatureShare(privateKeyBytes []byte, message []byte) ([]byte, error) {
+	privateKey, err := convertBytesToPrivateKey(privateKeyBytes, bms.keyGen)
 	if err != nil {
 		return nil, err
 	}
 
-	data.sigShares[data.ownIndex] = sigShareBytes
-
-	return sigShareBytes, nil
-}
-
-// not concurrent safe, should be used under RLock mutex
-func (bms *blsMultiSigner) isIndexInBitmap(index uint16, bitmap []byte) error {
-	indexOutOfBounds := index >= uint16(len(bms.data.pubKeys))
-	if indexOutOfBounds {
-		return crypto.ErrIndexOutOfBounds
-	}
-
-	indexNotInBitmap := bitmap[index/8]&(1<<uint8(index%8)) == 0
-	if indexNotInBitmap {
-		return crypto.ErrIndexNotSelected
-	}
-
-	return nil
+	return bms.llSigner.SignShare(privateKey, message)
 }
 
 // VerifySignatureShare verifies the single signature share of the signer with specified position
 // Signature is verified over a message configured with a previous call of SetMessage
-func (bms *blsMultiSigner) VerifySignatureShare(index uint16, sig []byte, message []byte, _ []byte) error {
+func (bms *blsMultiSigner) VerifySignatureShare(publicKey []byte, message []byte, sig []byte) error {
 	if sig == nil {
 		return crypto.ErrNilSignature
 	}
 
-	bms.mutSigData.RLock()
-	defer bms.mutSigData.RUnlock()
-
-	indexOutOfBounds := index >= uint16(len(bms.data.pubKeys))
-	if indexOutOfBounds {
-		return crypto.ErrIndexOutOfBounds
-	}
-
-	pubKey := bms.data.pubKeys[index]
-
-	return bms.llSigner.VerifySigShare(pubKey, message, sig)
-}
-
-// StoreSignatureShare stores the partial signature of the signer with specified position
-// Function does not validate the signature, as it expects caller to have already called VerifySignatureShare
-func (bms *blsMultiSigner) StoreSignatureShare(index uint16, sig []byte) error {
-	err := bms.llSigner.VerifySigBytes(bms.keyGen.Suite(), sig)
```

Comment:
here, we remove the `VerifySigBytes` since it's not needed here anymore, but in the signing component in elrond-go (for example) we might need to do this check over the signature. Can we maybe expose this function from here to access it from high level multisigner? I don't think it's exposed right now via other interfaces